core/src/main/java/org/mule/serialization/internal/NodeWrapper.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.serialization.internal;
+
+import org.w3c.dom.Attr;
+import org.w3c.dom.DOMException;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.UserDataHandler;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.stream.XMLStreamWriter;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.io.StringWriter;
+import java.io.WriteAbortedException;
+
+public class NodeWrapper implements Serializable, Node {
+  public static final String NODE_WRAPPER_ENVELOPE_OPEN_TAG = "<node-wrapper-envelope>";
+  public static final String NODE_WRAPPER_ENVELOPE_CLOSE_TAG = "</node-wrapper-envelope>";
+  public static final String CDATA_OPEN = "<![CDATA[";
+  public static final String CDATA_CLOSE = "]]>";
+  private final short nodeType;
+  private final String nodeName;
+  private transient Node node;
+  private String nodeAsString;
+
+  public NodeWrapper(Node node) {
+    this.node = node;
+    this.nodeType = node.getNodeType();
+    this.nodeName = node.getNodeName();
+  }
+
+  private Object writeReplace() throws ObjectStreamException {
+
+    try {
+        // In tests this if is not needed but because the Transformer Implementation changes in the actual standalone
```

Comment:
Remove the but "needed but because"

Reply:
Changed it to: "This ensures the CDATA is not transformed into a text node"